<?xml version="1.0"?>

<launch>
  <arg name="model_name" default="faster_rcnn_inception_v2_coco_2018_01_28"/>
  <arg name="num_classes" default="90"/>
  <arg name="label_file" default="mscoco_label_map.pbtxt"/>
  <arg name="camera_topic" default="/camera/color/image_raw"/>
  <arg name="depth_topic" default="/camera/aligned_depth_to_color/image_raw"/>
  <arg name="cloud_topic" default="/camera/depth/color/points"/>

  <node name="object_detection" pkg="ds_object_detection" ns="tensorflow_detection" type="object_detection.py" output="screen">
    <param name="model_name" value="$(arg model_name)"/>
    <param name="num_classes" value="$(arg num_classes)"/>
    <param name="label_file" value="$(arg label_file)"/>
    <param name="camera_topic" value="$(arg camera_topic)"/>
    <param name="depth_topic" value="$(arg depth_topic)"/>
    <param name="cloud_topic" value="$(arg cloud_topic)"/>
    <param name="num_workers" value="-1"/>  
  </node>
</launch>
